src/components/modal/report/modal.jsx

```diff
@@ -0,0 +1,186 @@
+const bindAll = require('lodash.bindall');
+const PropTypes = require('prop-types');
+const React = require('react');
+const FormattedMessage = require('react-intl').FormattedMessage;
+const injectIntl = require('react-intl').injectIntl;
+const intlShape = require('react-intl').intlShape;
+const Modal = require('../base/modal.jsx');
+const log = require('../../../lib/log.js');
+
+const Form = require('../../forms/form.jsx');
+const Button = require('../../forms/button.jsx');
+const Select = require('../../forms/select.jsx');
+const Spinner = require('../../spinner/spinner.jsx');
+const TextArea = require('../../forms/textarea.jsx');
+
+require('../../forms/button.scss');
+require('./modal.scss');
+
+class ReportModal extends React.Component {
+    constructor (props) {
+        super(props);
+        bindAll(this, [
+            'handleReasonSelect',
+            'handleSubmit'
+        ]);
+        this.state = {
+            prompt: props.intl.formatMessage({id: 'report.promptPlaceholder'}),
+            reason: '',
+            waiting: false
+        };
+    }
+    handleReasonSelect (name, value) {
+        const prompts = [
+            this.props.intl.formatMessage({id: 'report.promptCopy'}),
+            this.props.intl.formatMessage({id: 'report.promptUncredited'}),
+            this.props.intl.formatMessage({id: 'report.promptScary'}),
+            this.props.intl.formatMessage({id: 'report.promptLanguage'}),
+            this.props.intl.formatMessage({id: 'report.promptMusic'}),
+            this.props.intl.formatMessage({id: 'report.promptPersonal'}),
+            this.props.intl.formatMessage({id: 'report.promptGuidelines'}),
+            'not used',
+            this.props.intl.formatMessage({id: 'report.promptImage'})
+        ];
+        this.setState({prompt: prompts[value], reason: value});
+    }
+    handleSubmit (formData) {
+        this.setState({waiting: true});
+        this.props.onReport(formData, err => {
```

Comment:
If they came in as props, then we wouldn't need to pass in a callback, it could all be managed in the parent component.